crux-metrics/src/crux/metrics/dropwizard/console.clj

```diff
@@ -1,33 +1,19 @@
 (ns crux.metrics.dropwizard.console
   "Console reporting"
-  (:import java.util.concurrent.TimeUnit
-           [com.codahale.metrics MetricRegistry ScheduledReporter ConsoleReporter Clock MetricFilter]
-           java.io.PrintStream
-           java.util.Locale))
+  (:require [crux.metrics.dropwizard :as dropwizard])
+  (:import [com.codahale.metrics MetricRegistry ConsoleReporter]))
 
-(defn ^com.codahale.metrics.ConsoleReporter reporter
-  [^MetricRegistry reg opts]
-  (let [b (ConsoleReporter/forRegistry reg)]
-    (when-let [^PrintStream s (:stream opts)]
-      (.outputTo b s))
-    (when-let [^Locale l (:locale opts)]
-      (.formattedFor b l))
-    (when-let [^Clock c (:clock opts)]
-      (.withClock b c))
-    (when-let [^TimeUnit ru (:rate-unit opts)]
-      (.convertRatesTo b ru))
-    (when-let [^TimeUnit du (:duration-unit opts)]
-      (.convertDurationsTo b du))
-    (when-let [^MetricFilter f (:filter opts)]
-      (.filter b f))
-    (.build b)))
+(defn reporter ^ConsoleReporter
+  [^MetricRegistry reg {::keys [stream locale clock rate-unit duration-unit metric-filter]}]
+  (.build
+    (cond-> (ConsoleReporter/forRegistry reg)
+      stream (.outputTo stream)
+      locale (.formattedFor locale)
+      clock (.withClock clock)
+      rate-unit (.convertRatesTo rate-unit)
+      duration-unit (.convertDurationsTo duration-unit)
+      metric-filter (.filter metric-filter))))
 
-(defn start
-  "Report all metrics to standard out periodically"
-  [^ScheduledReporter r ^long seconds]
-  (.start r seconds))
-
-(defn stop
-  "Stops reporting all metrics to standard out."
-  [^ScheduledReporter r]
-  (.stop r))
+(defn start-reporter
```

Comment:
given that every caller of `reporter` is likely going to call `start-reporter`, I would probably merge these together